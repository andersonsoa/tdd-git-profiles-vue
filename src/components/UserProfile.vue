<template>
  <div class="user-profile__container">
    <img :src="user.avatar_url" class="user-profile__avatar" />
    <h1 class="user-profile__name">{{ user.name }}</h1>
    <p class="user-profile__bio">{{ user.bio }}</p>
  </div>
</template>

<script>
export default {
  name: "UserProfile",
  props: {
    user: {
      require: true,
      type: Object
    }
  }
};
</script>

<style scoped>
.user-profile__container {
  width: 100%;
  height: 500px;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
}
.user-profile__avatar {
  width: 300px;
  height: 300px;
  border-radius: 50%;
  border: 10px solid black;
  box-shadow: 1px 1px 10px #0004;
}
.user-profile__bio {
  width: 80%;
}
</style>
