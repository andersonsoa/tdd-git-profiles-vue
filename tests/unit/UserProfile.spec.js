import { shallowMount } from "@vue/test-utils";
import UserProfile from "@/components/UserProfile";
import user from "./fixtures/user";

describe("UserProfile", () => {
  let props;

  const build = () => {
    const wrapper = shallowMount(UserProfile, {
      propsData: props
    });
    return {
      wrapper,
      avatar: () => wrapper.find(".user-profile__avatar"),
      name: () => wrapper.find(".user-profile__name"),
      bio: () => wrapper.find(".user-profile__bio")
    };
  };

  beforeEach(() => {
    props = {
      user
    };
  });

  it("renderiza o componente", () => {
    // arrange
    const { wrapper } = build();
    debugger;
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renderiza os componentes principais", () => {
    // arrange
    const { avatar, name, bio } = build();
    // assert
    expect(avatar().exists()).toBe(true);
    expect(avatar().attributes().src).toBe(props.user.avatar_url);

    expect(name().exists()).toBe(true);
    expect(name().text()).toBe(props.user.name);

    expect(bio().exists()).toBe(true);
    expect(bio().text()).toBe(props.user.bio);
  });
});
