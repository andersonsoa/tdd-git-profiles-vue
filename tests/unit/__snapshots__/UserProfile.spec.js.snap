// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renderiza o componente 1`] = `
<div class="user-profile__container"><img src="https://avatars3.githubusercontent.com/u/22595773?v=4" class="user-profile__avatar">
  <h1 class="user-profile__name">Octocat Jedi</h1>
  <p class="user-profile__bio">These are not the octocats you're looking for...</p>
</div>
`;
